File: binding/src/libqfloat16.cpp
#include <qfloat16.h>
#include "libqfloat16.h"
#include "libqfloat16.hxx"

qfloat16* qfloat16_new() {
	 return new qfloat16();
}

qfloat16* qfloat16_new2(int param1) {
	 return new qfloat16(static_cast<Qt::Initialization>(param1));
}

qfloat16* qfloat16_new3(float f) {
	 return new qfloat16(f);
}

void qfloat16_CopyAssign(qfloat16* self, qfloat16* other) {
    *self = *other;
}

void qfloat16_MoveAssign(qfloat16* self, qfloat16* other) {
    *self = std::move(*other);
}

float qfloat16_Operatorfloat(const qfloat16* self) {
	return self->operator float();
}

bool qfloat16_IsInf(const qfloat16* self) {
	return self->isInf();
}

bool qfloat16_IsNaN(const qfloat16* self) {
	return self->isNaN();
}

bool qfloat16_IsFinite(const qfloat16* self) {
	return self->isFinite();
}

int qfloat16_FpClassify(const qfloat16* self) {
	return self->fpClassify();
}

bool qfloat16_IsNormal(const qfloat16* self) {
	return self->isNormal();
}

void qfloat16_Delete(qfloat16* self) {
    delete self;
}

File: binding/src/libqfloat16.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qfloat16.html
pub const qfloat16 = struct {

    /// New constructs a new qfloat16 object.
    pub fn New(other: ?*anyopaque) QtC.qfloat16 {
        return qtc.qfloat16_new(@ptrCast(other));
    }


    /// New2 constructs a new qfloat16 object.
    pub fn New2(other: ?*anyopaque) QtC.qfloat16 {
        return qtc.qfloat16_new2(@ptrCast(other));
    }


    /// New3 constructs a new qfloat16 object.
    pub fn New3() QtC.qfloat16 {
        return qtc.qfloat16_new3();
    }


    /// New4 constructs a new qfloat16 object.
    pub fn New4(param1: i32) QtC.qfloat16 {
        return qtc.qfloat16_new4(@intCast(param1));
    }


    /// New5 constructs a new qfloat16 object.
    pub fn New5(f: f32) QtC.qfloat16 {
        return qtc.qfloat16_new5(f);
    }


    /// New6 constructs a new qfloat16 object.
    pub fn New6(param1: ?*anyopaque) QtC.qfloat16 {
        return qtc.qfloat16_new6(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.qfloat16_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.qfloat16_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Operatorfloat(self: ?*anyopaque, ) f32 {
        return qtc.qfloat16_Operatorfloat(@ptrCast(self));
    }

    pub fn IsInf(self: ?*anyopaque, ) bool {
        return qtc.qfloat16_IsInf(@ptrCast(self));
    }

    pub fn IsNaN(self: ?*anyopaque, ) bool {
        return qtc.qfloat16_IsNaN(@ptrCast(self));
    }

    pub fn IsFinite(self: ?*anyopaque, ) bool {
        return qtc.qfloat16_IsFinite(@ptrCast(self));
    }

    pub fn FpClassify(self: ?*anyopaque, ) i32 {
        return qtc.qfloat16_FpClassify(@ptrCast(self));
    }

    pub fn CopySign(self: ?*anyopaque, sign: QtC.qfloat16) QtC.qfloat16 {
        return qtc.qfloat16_CopySign(@ptrCast(self), @ptrCast(sign));
    }

    pub fn LimitEpsilon() QtC.qfloat16 {
        return qtc.qfloat16_LimitEpsilon();
    }

    pub fn LimitMin() QtC.qfloat16 {
        return qtc.qfloat16_LimitMin();
    }

    pub fn LimitDenormMin() QtC.qfloat16 {
        return qtc.qfloat16_LimitDenormMin();
    }

    pub fn LimitMax() QtC.qfloat16 {
        return qtc.qfloat16_LimitMax();
    }

    pub fn LimitLowest() QtC.qfloat16 {
        return qtc.qfloat16_LimitLowest();
    }

    pub fn LimitInfinity() QtC.qfloat16 {
        return qtc.qfloat16_LimitInfinity();
    }

    pub fn LimitQuiet_NaN() QtC.qfloat16 {
        return qtc.qfloat16_LimitQuiet_NaN();
    }

    pub fn LimitSignaling_NaN() QtC.qfloat16 {
        return qtc.qfloat16_LimitSignaling_NaN();
    }

    pub fn IsNormal(self: ?*anyopaque, ) bool {
        return qtc.qfloat16_IsNormal(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.qfloat16_Delete(@ptrCast(self));
    }
};
